### PR TITLE
PR: Update Pylint plugin to Python 3

### LINKED ISSUE #11725: Move PyLint plugin to Python 3
Take into account the following:

- [ ] Update `super()` use
- [ ] Remove the `py3compat` module
- [ ] Update any code (and remove if necessary) that uses `if PY2` etc...
- [ ] Check division operator
- [ ] Remove future imports
- [ ] Check tests
- [ ] Check for Py2 of Py3 codes

### PR DESCRIPTION
**Pull request by steff456**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

Update the code of the pylint plugin to python 3

### Issue(s) Resolved

Fixes #11725 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: Steff456

### CODE REVIEW

File: spyder/plugins/pylint/widgets/pylintgui.py [hunk 1]
```diff
@@ -182,17 +186,20 @@ def __init__(self, parent, max_entries=100, options_button=None,
...
-                                    text=_("Output"),
-                                    text_beside_icon=True,
-                                    tip=_("Complete output"),
-                                    triggered=self.show_log)
+        self.log_button = create_toolbutton(self,
+                                            icon=ima.icon('log'),
+                                            text=_("Output"),
+                                            text_beside_icon=True,
+                                            tip=_("Complete output"),
+                                            triggered=self.show_log)
```
Comment (ccordoba12) [inline spyder/plugins/pylint/widgets/pylintgui.py]: Let's format these lines in the same way as the other buttons above:

```python
self.log_button = create_toolbutton(
    self,
    icon=ima.icon('log'),
    ...)
```

File: spyder/plugins/pylint/widgets/pylintgui.py [hunk 2]
```diff
@@ -499,9 +505,7 @@ def show_data(self, justanalyzed=False):
                     text_prun = ' (%s %s/10)' % (text_prun, previous_rate)
                     text += prevrate_style % (self.prevrate_color, text_prun)
                 self.treewidget.set_results(filename, results)
-                date = to_text_string(time.strftime("%Y-%m-%d %H:%M:%S",
-                                                    datetime),
-                                      encoding='utf8')
```
Comment (ccordoba12) [inline spyder/plugins/pylint/widgets/pylintgui.py]: Why there's no need to convert this string to `utf8`?

Comment (steff456) [reply spyder/plugins/pylint/widgets/pylintgui.py]: Because the time.strftime returns a string, so there's no need to cast it

Comment (ccordoba12) [reply spyder/plugins/pylint/widgets/pylintgui.py]: Ok, nice! :+1:

**Pull request merged**